test/games/strategy/triplea/delegate/StratBombTest.java

```diff
@@ -0,0 +1,116 @@
+    return GameDataTestUtil.getDelegateBridge(player, m_data);
+  }
+
+
+  @Test
+  public void testBombingRaid2targets() {
+    final Territory germany = m_data.getMap().getTerritory("Germany");
+    final Territory uk = m_data.getMap().getTerritory("United Kingdom");
+    final PlayerID germans = GameDataTestUtil.germans(m_data);
+    final PlayerID british = GameDataTestUtil.british(m_data);
+    // add a unit
+    final Unit stratBomber = GameDataTestUtil.bomber(m_data).create(british);
+    final Unit tacBomber1 = GameDataTestUtil.tacBomber(m_data).create(british);
+    final Unit tacBomber2 = GameDataTestUtil.tacBomber(m_data).create(british);
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(stratBomber)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber1)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber2)));
+    final BattleTracker tracker = new BattleTracker();
+    List<Unit> attackers = Collections.emptyList();
+    attackers.add(stratBomber);
+    attackers.add(tacBomber1);
+    attackers.add(tacBomber2);
+    HashMap<Unit, HashSet<Unit>> targets = null;
+   
+
+//    final Collection<Unit> enemyTargets = Match.getMatches(uk.getUnits().getMatches(new CompositeMatchAnd<>(Matches.enemyUnit(germans, m_data),              Matches.UnitIsAtMaxDamageOrNotCanBeDamaged(uk).invert(), Matches.unitIsBeingTransported().invert())), Matches.UnitIsLegalBombingTargetBy(stratBomber));
+
+    for( final Unit target : germany.getUnits().getUnits() ) {
+    	System.out.println(target.getType().getName());
+    	switch( target.getType().getName() ) {
+				case "airfield"      : targets.put(target, new HashSet<>(Collections.singleton(tacBomber1)));
+					break;
+				case "harbour"       : targets.put(target, new HashSet<>(Collections.singleton(tacBomber2)));
+					break;
+				case "factory_major" : targets.put(target, new HashSet<>(Collections.singleton(stratBomber)));
+					break;
+		  }
+		}
+    final ITestDelegateBridge bridge = getDelegateBridge(british);
+    tracker.addBattle(new RouteScripted(germany), attackers, true, british, bridge, null, null, targets, true);
+                
+    final StrategicBombingRaidBattle battle = new StrategicBombingRaidBattle(germany, m_data, british, tracker);
+    battle.addAttackChange(m_data.getMap().getRoute(uk, germany), uk.getUnits().getMatches(Matches.UnitIsStrategicBomber), null);
+    addTo(germany, uk.getUnits().getMatches(Matches.UnitIsStrategicBomber));
+    tracker.getBattleRecords(m_data).addBattle(british, battle.getBattleID(), germany, battle.getBattleType(), m_data);
+    // aa guns rolls 1,3, first one hits, remaining bomber rolls 1 dice at 2
+    bridge.setRandomSource(new ScriptedRandomSource(new int[] {3, 3, 2}));
+    // if we try to move aa, then the game will ask us if we want to move
+    // fail if we are called
+    final InvocationHandler handler = new InvocationHandler() {
```

Comment:
You should replace this with a lambda and merge into line 106:... ITripleAPlayer.class), (proxy, method, args) -> {return null;});